examples/models/object_detection/faster_rcnn/train.py

```diff
@@ -66,194 +66,75 @@
+    image = inputs["image"]
+    image = tf.cast(image, tf.float32)
+    image = tf.keras.applications.resnet50.preprocess_input(image)
+    gt_boxes = inputs["objects"]["bbox"]
+    image, gt_boxes = flip_fn(image, gt_boxes)
+    gt_boxes = keras_cv.bounding_box.convert_format(
+        gt_boxes,
+        images=image,
+        source="rel_yxyx",
+        target=bounding_box_format,
     )
-    indices = get_non_empty_box_indices(gt_boxes)
-    gt_boxes = tf.gather(gt_boxes, indices)
-    gt_classes = tf.gather(gt_classes, indices)
-    return image, gt_boxes, gt_classes
-
-
-def flip_fn(image, boxes):
-    if tf.random.uniform([], minval=0, maxval=1, dtype=tf.float32) > 0.5:
-        image = tf.image.flip_left_right(image)
-        y1, x1, y2, x2 = tf.split(boxes, num_or_size_splits=4, axis=-1)
-        boxes = tf.concat([y1, 1.0 - x2, y2, 1.0 - x1], axis=-1)
-    return image, boxes
-
+    gt_classes = tf.cast(inputs["objects"]["label"], tf.float32)
+    image, gt_boxes, gt_classes = resize_fn(image, gt_boxes, gt_classes)
+    gt_classes = tf.expand_dims(gt_classes, axis=-1)
 
-def proc_train_fn(bounding_box_format, img_size):
-    def apply(inputs):
-        image = inputs["image"]
-        image = tf.cast(image, tf.float32)
-        image = tf.keras.applications.resnet50.preprocess_input(image)
-        gt_boxes = inputs["objects"]["bbox"]
-        image, gt_boxes = flip_fn(image, gt_boxes)
-        gt_boxes = keras_cv.bounding_box.convert_format(
-            gt_boxes,
-            images=image,
-            source="rel_yxyx",
-            target=bounding_box_format,
-        )
-        gt_classes = tf.cast(inputs["objects"]["label"], tf.float32)
-        image, gt_boxes, gt_classes = resize_fn(image, gt_boxes, gt_classes)
-        gt_classes = tf.expand_dims(gt_classes, axis=-1)
+    gt_boxes = tf.concatenate([gt_boxes, gt_classes], axis=-1)
+    return {
+        "images": image,
+        "bounding_boxes": gt_boxes,
+    }
 
-        return {
-            "images": image,
-            "gt_boxes": gt_boxes,
-            "gt_classes": gt_classes,
-        }
 
-    return apply
-
-
-def pad_fn(examples):
-    gt_boxes = examples.pop("gt_boxes")
-    gt_classes = examples.pop("gt_classes")
+def pad_and_unpackage(examples):
+    boxes = examples.pop("bounding_boxes")
+    gt_boxes = boxes[:, :4]
+    gt_classes = boxes[:, 4:]
     gt_boxes = gt_boxes.to_tensor(default_value=-1.0, shape=[global_batch, 32, 4])
     gt_classes = gt_classes.to_tensor(default_value=-1.0, shape=[global_batch, 32, 1])
     return examples["images"], {"gt_boxes": gt_boxes, "gt_classes": gt_classes}
 
 
+train_ds = train_ds.apply(
+    tf.data.experimental.dense_to_ragged_batch(global_batch, drop_remainder=True)
+)
+
+augmenter = layers.Augmenter(
+    [
+        layers.RandomFlip(mode="horizontal", bounding_box_format="xywh"),
+        layers.JitteredResize(
+            desired_size=(640, 640),
+            padded_size=(640, 640),
```

Comment:
Should this just output ragged tensors and skip padding, then rely on keras_cv.layers.Resizing(pad_only=True)?